src/main/java/com/flipkart/zjsonpatch/JsonPatch.java

```diff
@@ -131,7 +131,7 @@ public static void validate(JsonNode patch) throws InvalidJsonPatchException {
     }
 
     public static JsonNode apply(JsonNode patch, JsonNode source, EnumSet<CompatibilityFlags> flags) throws JsonPatchApplicationException {
-        ApplyProcessor processor = new ApplyProcessor(source);
+        InPlaceApplyProcessor processor = new InPlaceApplyProcessor(source);
```

Comment:
Shouldn't this be
CopyingApplyProcessor processor = new CopyingApplyProcessor(source);
Lets have deepCopy() as default behaviour, Since as per RFC source json should not be modified.
Hence, after your chance, we will have 2 public APIs, app() & applyInPlace().